### ISSUE TITLE
Optimize FDT validation main loop with JAX JIT compilation

### ISSUE THREAD

**Issue by anjor** [anjor/gandalf#67]
## Overview
Extract and JIT-compile the main evolution loop in `run_forced_single_mode()` for 2-5× speedup in FDT validation tests.

## Current Status
Main loop in `validation.py:336-367` is **not JIT-compiled**, causing repeated overhead:
- Python loop control
- Function call overhead for `force_alfven_modes()` and `gandalf_step()`
- Array allocation/copying

Current runtime: **~13 seconds** for 250 steps (M=10, 32×32×16 grid)

### 1. Extract Loop Body into Pure Function
```python
@jax.jit
def _fdt_evolution_step(state, key, dt, forcing_params, physics_params):
    """Single evolution step: forcing + dynamics."""
    # Apply forcing
    state, key = force_alfven_modes(state, key=key, **forcing_params)
    
    # Evolve dynamics
    state = gandalf_step(state, dt=dt, **physics_params)
    
    return state, key
```

### 2. Use jax.lax.scan for Loop
```python
def carry_fn(carry, step_idx):
    state, key = carry
    state, key = _fdt_evolution_step(state, key, dt, forcing_params, physics_params)
    return (state, key), None

(state, key), _ = jax.lax.scan(carry_fn, (state, key), jnp.arange(n_steps))
```

### 3. Separate Warmup and Averaging Phases
- Warmup: Fast scan without spectrum accumulation
- Averaging: Scan with spectrum collection via `scan` output

## Expected Performance Gains

**Conservative estimate:** 2-3× speedup
- Eliminates Python loop overhead
- Better JAX optimization (loop fusion, memory reuse)
- Single compilation for entire evolution

**Optimistic estimate:** 4-5× speedup if memory layout optimized

**Target runtime:** ~3-5 seconds (down from 13s)

### Challenges
1. **State accumulation:** Need to collect spectrum at each step after warmup
   - Solution: Use `scan` output to accumulate spectra, then average in numpy

2. **NaN/Inf checks:** Currently check every step (lines 351-357)
   - Solution: Check only at end, or use `jax.debug.callback` for monitoring

3. **Random key splitting:** `force_alfven_modes` advances RNG state
   - Solution: Thread key through scan carry

### Benefits
- Faster test suite (3 tests: ~40s → ~12-15s total)
- Enables more comprehensive parameter sweeps
- Scales better to larger grids and longer runs

## Testing
- Verify identical results (deterministic RNG seeding)
- Benchmark before/after with various grid sizes
- Profile with JAX profiler to identify bottlenecks

## Priority
**Medium** - Nice to have, not blocking production use

Current performance acceptable for testing (13s per run). Optimization worthwhile if:
- Running large parameter sweeps
- Need faster CI/CD feedback
- Exploring finer grids or more moments

## Related
- PR #65: Current FDT validation implementation
- Issue #66: Exact FDT equations (higher priority)
- Original code review: Noted JIT opportunity for 20-30% speedup
